### DISCUSSION TITLE
Get real file system song path from API

### DISCUSSION THREAD

**Discussion by daniele-athome** [navidrome/navidrome#3203] [Q&A] [answered]
Hello there,
I can see that the Subsonic API doesn't return the real file system path of songs. Was that decision by design? Can I ask the reasoning behind that? I'm asking because I'd like to know if it would make sense to contribute something to make that happen.

Also I was wondering if there is another API to access that information (maybe the ones used by the web app, but I can't find any documentation, I guess is not really a public API?). In my case I would need to "dump" Navidrome playlists into M3U files (with a custom script) so I would need the real file system paths of the songs.

Of course there is also reading from the database directly, but I would consider that a last resort.

Thanks!

EDIT: I see there is a command for exporting a playlist, but it requires executing a command inside the Navidrome container, but that would require privileges to execute Docker and some other caveats... another last resort.

**Comment by deluan** [comment, marked as answer]
> Was that decision by design?

Yes, to match the decision of not having real browse-by-folder endpoints. If the paths didn't match, some clients would break.

> I guess is not really a public API

No, it is not (yet)

Anyway, you can enable Real Paths per client/player:
![Screenshot 2024-08-21 at 10 41 23 AM](https://github.com/user-attachments/assets/562c7c49-b854-4fbc-9993-b3c70260d40c)

**Reply by daniele-athome** [reply]
> Anyway, you can enable Real Paths per client/player:

I've never noticed that option before, thank you!
